src/Bicep.Core/Emit/ExpressionConverter.cs

```diff
@@ -245,7 +248,7 @@ private LanguageExpression ConvertAccessExpression(AccessExpression expression,
                 }
                 else
                 {
-                    properties = properties.Append(new JTokenExpression("value"));
+                    properties = isSecureOutput ? properties : properties.Append(new JTokenExpression("value"));
```

Comment:
You'll need to update line 247 with a similar ternary (to `properties = isSecureOutput ? Enumerable.Empty<LanguageExpression>() : new[] { new JTokenExpression("value") };` or the like)